src/test/lrsql/admin/protocol_test.clj

```diff
@@ -155,7 +187,79 @@
+        (testing "delete-actor correctly deletes statements that are parent to actor (sub)statements"
+          (lrsp/-store-statements lrs auth-ident [stmt-3] [])
+          (is (= 1 (count-of-actor parent-ifi)))
+          (adp/-delete-actor lrs {:actor-ifi child-ifi})
+          (is (zero? (count-of-actor parent-ifi))) ;
+          (adp/-delete-actor lrs {:actor-ifi parent-ifi}))
+        (testing "delete-actor correctly deletes substatements that are child to actor statements"
+          (lrsp/-store-statements lrs auth-ident [stmt-3] [])
+          (is (= 1 (count-of-actor child-ifi)))
+          (adp/-delete-actor lrs {:actor-ifi parent-ifi})
+          (is (zero? (count-of-actor child-ifi)))
+          (adp/-delete-actor lrs {:actor-ifi child-ifi}))
+        (testing "for StatementRefs, delete-actor deletes statement->actor relationships but leaves statements by another actor untouched"
+          (let [[ifi-0 ifi-2] (mapv stmt->ifi [stmt-0 stmt-2])]
+            (lrsp/-store-statements lrs auth-ident [stmt-0 stmt-2] [])
+            (is (= 2 (count-of-actor ifi-0)))
+            (is (= 2 (count-of-actor ifi-2)))
+            (adp/-delete-actor lrs {:actor-ifi ifi-0})
+            (is (= 1 (count-of-actor ifi-0)))
+            (is (= 1 (count-of-actor ifi-2)))
+            (adp/-delete-actor lrs {:actor-ifi ifi-2})
+            (is (zero? (count-of-actor ifi-0)))
+            (is (zero? (count-of-actor ifi-2)))))))
+
+    (testing "delete-actor handles dependent tables: state_document"
+      (let [ifi (ua/actor->ifi (:agent lrst/state-id-params))
+            ds (-> lrs :connection :conn-pool)]
+      (lrsp/-set-document lrs auth-ident lrst/state-id-params lrst/state-doc-1 true)
+      (adp/-delete-actor lrs {:actor-ifi ifi})
+      (is (empty? (jdbc/execute! ds ["select * from state_document where agent_ifi  = ?" ifi])))))
+
+    (testing "delete-actor handles dependent tables: agent_profile_document"
+      (let [ifi (ua/actor->ifi (:agent lrst/agent-prof-id-params))
+            ds (-> lrs :connection :conn-pool)]
+        (lrsp/-set-document lrs auth-ident lrst/agent-prof-id-params lrst/agent-prof-doc true)
+        (adp/-delete-actor lrs {:actor-ifi ifi})
+        (is (empty? (jdbc/execute! ds ["select * from agent_profile_document where agent_ifi  = ?" ifi])))))
+
+    (let [ifi (ua/actor->ifi (lrst/stmt-4 "actor"))
+          stmt-id (u/str->uuid (lrst/stmt-4 "id"))
+          ds (-> lrs :connection :conn-pool)]
+      (lrsp/-store-statements lrs auth-ident [lrst/stmt-4] [lrst/stmt-4-attach])
+      (adp/-delete-actor lrs {:actor-ifi ifi})
+
+      (testing "delete-actor handles statement attachments"
+        (is (empty? (jdbc/execute! ds ["select * from attachment where statement_id  = ?" stmt-id]))))
+      (testing "delete actor handles intermediate statement-to-activity table"
+        (is (empty? (jdbc/execute! ds ["select * from statement_to_activity where statement_id  = ?" stmt-id])))))
```

Comment:
Okay this I'll slide since there's no query function to query the join table itself. I would extract this part into a helper function though.